src/Framework/ContaoFramework.php

```diff
@@ -203,11 +205,11 @@ private function setConstants()
      */
     private function getMode()
     {
-        if ($this->isBackendScope()) {
+        if ($this->request && $this->isBackendScope($this->request)) {
             return 'BE';
         }
 
-        if ($this->isFrontendScope()) {
+        if ($this->request && $this->isFrontendScope($this->request)) {
```

Comment:
Nope, you will get an error that instance of Request is expected.
You will only get null handled in the depreacted trait.